src/cpp/xmlparser/XMLParser.cpp

```diff
@@ -881,15 +913,176 @@ p_dynamictypebuilder_t XMLParser::parseXMLMemberDynamicType(tinyxml2::XMLElement
+            //factory->DeleteBuilder(innerBuilder);
+        }
+    }
+    else if (memberMap != nullptr)
+    {
+        /*
+            In maps allowed formats are (complex format includes the basic):
+            map<map<long, long, 6>, map<long, map<long, short>,2>
+            <map name="my_map" length="2">
+                <key_type>
+                    <map name="inner_key_map" key_type="long" value_type="long" length="6"/>
+                </key_type>
+                </value_type>
+                    <map name="inner_value_map" key_type="long" length="2">
+                        </value_type>
+                            <map name="inner_value_value_map" key_type="long" value_type="short"/>
+                        </value_type>
+                    </map>
+                </value_type>
+            </map>
+            In this example, inner maps names are ignored and can be omited.
+         */
+        // Parse key
+
+        //const char* keyType = p_root->Attribute(KEY);
+        p_dynamictypebuilder_t keyTypeBuilder;
+        if (memberMapKeyType == nullptr)
+        {
+            //tinyxml2::XMLElement* keyElement = p_root->FirstChildElement(KEY);
+            //if (keyElement == nullptr)
+            //{
+                logError(XMLPARSER, "Error parsing key_type element: Not found.");
+                return nullptr;
+            //}
+            //keyTypeBuilder = parseXMLMemberDynamicType(keyElement->FirstChildElement(), nullptr, MEMBER_ID_INVALID);
+        }
+        else
+        {
+            keyTypeBuilder = getDiscriminatorTypeBuilder(memberMapKeyType);
+        }
+
+        if (keyTypeBuilder == nullptr)
+        {
+            logError(XMLPARSER, "Error parsing map's key element type: Cannot be recognized.");
+            return nullptr;
+        }
+
+        // Parse value
+        //const char* valueType = p_root->Attribute(VALUE_TYPE);
+        p_dynamictypebuilder_t valueTypeBuilder;
+        if (memberType == nullptr)
+        {
+            //tinyxml2::XMLElement* valueElement = p_root->FirstChildElement(VALUE_TYPE);
+            //if (valueElement == nullptr)
+            //{
+                logError(XMLPARSER, "Error parsing value_value element: Not found.");
+                return nullptr;
+            //}
+            //valueTypeBuilder = parseXMLMemberDynamicType(valueElement->FirstChildElement(), nullptr, MEMBER_ID_INVALID);
+        }
+        else
+        {
+            valueTypeBuilder = getDiscriminatorTypeBuilder(memberType);
+        }
+
+        if (valueTypeBuilder == nullptr)
+        {
+            logError(XMLPARSER, "Error parsing map's value element type: Cannot be recognized.");
+            return nullptr;
+        }
+
+        const char* lengthStr = p_root->Attribute(MAP_MAXLENGTH);
+        uint32_t length = MAX_ELEMENTS_COUNT;
+        if (lengthStr != nullptr)
+        {
+            length = std::stoi(lengthStr);
```

Comment:
Be careful, `lengthStr` could include expressions.

Reply:
For future features